System/Posix/IO/Common.hsc

```diff
@@ -301,10 +304,26 @@ handleToFd' h h_@Handle__{haType=_,..} = do
      -- converting a Handle into an Fd effectively means
      -- letting go of the Handle; it is put into a closed
      -- state as a result.
-     flushWriteBuffer h_
+
+     -- free the spare buffers
+     writeIORef haBuffers BufferListNil
+     writeIORef haCharBuffer noCharBuffer
+     writeIORef haByteBuffer noByteBuffer
```

Comment:
Doesn't this change lose any unwritten data in the write buffer?  It used to be flushed, but now is just dropped?

With `hClose` this happens before the call to `hClose_handle_ handle_` (after which the above code is modelled):
```
hClose_help :: Handle__ -> IO (Handle__, Maybe SomeException)
hClose_help handle_ =
  case haType handle_ of
      ClosedHandle -> return (handle_,Nothing)
      _ -> do mb_exc1 <- trymaybe $ flushWriteBuffer handle_ -- interruptible
                    -- it is important that hClose doesn't fail and
                    -- leave the Handle open (#3128), so we catch
                    -- exceptions when flushing the buffer.
              (h_, mb_exc2) <- hClose_handle_ handle_
              return (h_, if isJust mb_exc1 then mb_exc1 else mb_exc2)
```

Reply:
I'm really not too sure about this change. I think it's not necessary for 2.8.0.0, since people have lived with it for over a decade.

Reply:
If the `memory leak` can be safely avoided, I'm inclined to fix anyway, but replacing flush with just dropping the buffer is not correct.  Otherwise, I don't see any issues.

Reply:
@hs-viktor I'm not too familiar with this part of the code. Feel free to adjust this PR.